aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/ReadHandler.java

```diff
@@ -28,25 +28,41 @@
 
         DescribeLogGroupsResponse response = null;
         ListTagsLogGroupResponse tagsResponse = null;
-        try {
-            response = proxy.injectCredentialsAndInvokeV2(Translator.translateToReadRequest(model),
-                    ClientBuilder.getClient()::describeLogGroups);
+        ResourceModel modelFromReadResult = null;
+        String nextToken = null;
+        // Keep paginating until requested log group is found
+        do {
             try {
-                tagsResponse = proxy.injectCredentialsAndInvokeV2(Translator.translateToListTagsLogGroupRequest(model.getLogGroupName()),
-                        ClientBuilder.getClient()::listTagsLogGroup);
-            } catch (final CloudWatchLogsException e) {
-                if (Translator.ACCESS_DENIED_ERROR_CODE.equals(e.awsErrorDetails().errorCode())) {
-                    // fail silently, if there is no permission to list tags
-                    logger.log(e.getMessage());
-                } else {
-                    throw e;
+                response = proxy.injectCredentialsAndInvokeV2(Translator.translateToReadRequest(model, nextToken),
+                        ClientBuilder.getClient()::describeLogGroups);
+                if (tagsResponse == null) {
+                    try {
```

Comment:
I don't think this try statement has to be nested within the other try statement. why not just execute them sequentially?

Reply:
I believe the person who implemented tag support wanted to call `ListTagsLogGroup` api before we call `translateForRead` since the tagging response is needed in that function. I can move the try statement outside and call `ListTagsLogGroup` before we call  `DescribeLogGroups` but that will mean if there is a request failure it will throw with whatever message is thrown by `ListTagsLogGroup`